### PR TITLE
MINOR, DOCS : Fixed old and added new javadocs to org.apache.kafka.common.utils.Utils

### PR DESCRIPTION
**Pull request by Serwios**
### Committer Checklist (excluded from commit message)
- [X] Verify design and implementation 
- [X] Verify test coverage and CI build status
- [X] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by Serwios** [comment]
Hi, request for review @AndrewJSchofield

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java [hunk 1]
```diff
@@ -888,9 +892,12 @@ public FileVisitResult postVisitDirectory(Path path, IOException exc) throws IOE
     }
 
     /**
-     * Returns an empty collection if this list is null
-     * @param other
-     * @return
+     * Returns an empty list if the provided list is null, otherwise returns the list itself.
+     * <p>
+     * This method is useful for avoiding `NullPointerException` when working with potentially null lists.
```
Comment (AndrewJSchofield) [inline clients/src/main/java/org/apache/kafka/common/utils/Utils.java]: I'd prefer `{@code NullPointerException}`.

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java [hunk 2]
```diff
@@ -572,9 +572,13 @@ public static String formatAddress(String host, Integer port) {
...
-     * Formats a byte number as a human-readable String ("3.2 MB")
-     * @param bytes some size in bytes
-     * @return
+     * Formats a byte value into a human-readable string with an appropriate unit
+     * (e.g., "3.2 KB", "1.5 MB", "2.1 GB"). The format includes two decimal places.
+     *
+     * @param bytes the size in bytes
+     * @return a string representing the size with the appropriate unit (e.g., "3.2 KB", "1.5 MB").
+     *         If the value is negative, the input is returned as a string (e.g., "-500").
+     *         If the number is too large for the predefined units, the raw byte value is returned.
```
Comment (AndrewJSchofield) [inline clients/src/main/java/org/apache/kafka/common/utils/Utils.java]: The "raw byte value is returned" is a bit misleading I think. I suggest "the input is returned as a string", which is just the same as for a negative value.

**Review by AndrewJSchofield — CHANGES_REQUESTED** [review]
Thanks for the PR. A couple of minor comments.

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java (hunk 1 above)
Comment (Serwios) [reply clients/src/main/java/org/apache/kafka/common/utils/Utils.java]: Fixed

File: clients/src/main/java/org/apache/kafka/common/utils/Utils.java (hunk 2 above)
Comment (Serwios) [reply clients/src/main/java/org/apache/kafka/common/utils/Utils.java]: Fixed in next commit

**Review by AndrewJSchofield — APPROVED** [review]

**Pull request merged**